core/client/app/controllers/settings/general.js

```diff
@@ -110,6 +124,75 @@ export default Controller.extend(SettingsSaveMixin, {
 
         toggleUploadLogoModal() {
             this.toggleProperty('showUploadLogoModal');
+        },
+
+        validateFacebookUrl(userInput) {
+            let newUrl = userInput.trim();
+            let oldUrl = this.get('model.facebook');
+            let errMessage = '';
+
+            if (!newUrl) {
+                // Clear out the Facebook url
+                this.set('model.facebook', '');
+                return;
+            }
+
+            if (!newUrl.match(/(^https:\/\/www\.facebook\.com\/)(\S+)/g)) {
+                errMessage = 'The URL must be in a format like ' +
+                    'https://www.facebook.com/yourUsername';
+                this.get('model.errors').add('facebook', errMessage);
+                return;
+            }
+
+            // If new url didn't change, exit
+            if (!oldUrl && oldUrl.match(newUrl)) {
+                return;
+            }
+
+            // Validation complete
+            this.set('model.facebook', newUrl);
+
+            this.get('model').save().catch((errors) => {
+                this.showErrors(errors);
+                this.get('model').rollbackAttributes();
+            });
+        },
+
+        validateTwitterUrl(userInput) {
+            let newUrl = userInput.trim();
+            let oldUrl = this.get('model.twitter');
+            let errMessage = '';
+            let username = '@';
+
+            if (!newUrl) {
+                // Clear out the Twitter url
+                this.set('model.twitter', '');
+                return;
+            }
+
+            if (!newUrl.match(/(^https:\/\/twitter\.com\/)(\S+)/g)) {
+                errMessage = 'The URL must be in a format like ' +
+                    'https://twitter.com/yourUsername';
+                this.get('model.errors').add('twitter', errMessage);
+                return;
+            }
+
+            // If new url didn't change, exit
+            if (!oldUrl && oldUrl.match(newUrl)) {
+                return;
+            }
+
+            // Validation complete
+            // extract username from url and give save it in model incl. @
+            let [ , , , user] = newUrl.match(/https?:\/\/(www\.)?twitter\.com\/(#!\/)?@?([^\/]*)/);
```

Comment:
If you only care about the username it might make sense to use non-capturing groups (e.g. `(?:www\.)?`) to simplify the destructuring assignment.